datastore/types/datastore_search_param.go

```diff
@@ -52,9 +52,18 @@ func (dssp *DataStoreSearchParam) WriteTo(writable types.Writable) {
 	dssp.ResultRange.WriteTo(contentWritable)
 	dssp.ResultOption.WriteTo(contentWritable)
 	dssp.MinimalRatingFrequency.WriteTo(contentWritable)
-	dssp.UseCache.WriteTo(contentWritable)
-	dssp.TotalCountEnabled.WriteTo(contentWritable)
-	dssp.DataTypes.WriteTo(contentWritable)
+
+	if dssp.StructureVersion >= 1 {
```

Comment:
In the wiki it says at the end of the section for this type:
> In NEX version 4.0, the revision number was set back to 0 with no other changes.

So I think for all of these `if`s we'd want an extra condition of `||  libraryVersion.GreaterOrEqual("4.0.0")` 

You can grab the library version from the stream as done [here](https://github.com/PretendoNetwork/nex-protocols-go/blob/b04c3e393e901f2901503e40c25a16d4a4209f46/datastore/types/datastore_req_get_info.go#L24-L25)

Reply:
Ohh good catch!